xarray/testing/assertions.py

```diff
@@ -95,6 +95,18 @@ def assert_isomorphic(a: DataTree, b: DataTree, from_root: bool = False):
         raise TypeError(f"{type(a)} not of type DataTree")
 
 
+def align_dims(a, b, check_dims):
+    if check_dims not in ("strict", "transpose"):
+        raise ValueError(f"Invalid value for check_dims: {check_dims}")
+    if not isinstance(a, (Variable, DataArray, Dataset)):
+        return b
+    if check_dims == "transpose":
+        assert set(a.dims) == set(b.dims), f"Dimensions differ: {a.dims} {b.dims}"
```

Comment:
Assuming that `Dataset.dims` returns an ordered type may no longer be true after #8980, but that's my problem, not yours 😅

Reply:
This future change is also raising a warning during the tests, not sure if I should be suppressing it.

Reply:
Okay so we should deal with it now then. Basically `Dataset` objects don't really have ordered dimensions, and `Dataset.dims` will be changed to return a `set` type in future. So `check_dim_order` can't do anything when comparing dataset objects, the most you can do is check that all the same dims are present.